elastic/kibana
# [Security Solution][Notes] - note icon to all tables and timeline

## Summary

This PR continues to add functionality to the new notes feature for Security Solution. It focuses on adding a note icon to the alerts table.

- add a note icon to a few tables in the application:
  - alerts table on the alerts page
  - alerts table on the cases page
  - events table in the hosts page
  - events table in the network page
  - events table in the users page
  - all timeline screens, with the `unifiedComponentsInTimelineEnabled` on and off
- add a red dot the the notes icon in the tables mentioned above as well as the timeline tables, signifying when an document has notes associated with it

When the table is loaded, a call is made to fetch all the notes for the documents visible on that page.

### TODO

- [x] add similar logic for the events table
- [x] verify that things work correctly in timeline

https://github.com/elastic/kibana/assets/17276605/f7ce26ab-b344-464f-bca1-369b70c72503

### How to test

- make sure the feature flag is enabled in your `kibana.yml` file: `xpack.securitySolution.enableExperimental: ['securitySolutionNotesEnabled']`
- verify that the note icon appears on all the tables mentioned above as well as timeline
- verify that when creating a note for an alert or event, the note icon reflect that in all other places for the same document id
- verify that no changes should be visible if either `securitySolutionNotesEnabled` or `expandableFlyoutDisabled` are on

_Note: as the server side code isn't ready, the frontend is mocking the fectch notes call and always returns one note per document id. If you load the alerts page in this PR, it is expected to see one note for each row of the table._

![Screenshot 2024-06-26 at 10 36 53 AM](https://github.com/elastic/kibana/assets/17276605/a4b25b4a-36c5-48f3-a2c1-d40f751c6e6f)

https://github.com/elastic/security-team/issues/9375

### Checklist
